src/lib/ares_android.c

```diff
@@ -80,6 +81,25 @@ static jmethodID jni_get_method_id(JNIEnv *env, jclass cls,
   return mid;
 }
 
+static const char *get_thread_name()
+{
+    static __thread char name[17] = {0};
```

Comment:
not a big fan of using ThreadLocalStorage like this unless its necessary.  I can't find in the docs if the name you pass is required to exist for the life of the thread or not.  If it does, then so be it, its necessary.

Otherwise, really, you should just create a function like: `int jvm_attach(JNIEnv *env);`
That does both the `get_thread_name()` and `get_jvm_attach_args()` internally as one function along with calling `return (*android_jvm)->AttachCurrentThread(android_jvm, env, &args);` or whatever.  Replacing all direct calls to AttachCurrentThread with that new function.

Reply:
Refactored